great_expectations/rule_based_profiler/rule_based_profiler.py

```diff
@@ -309,6 +313,39 @@ def run(
 
         return expectation_suite
 
+    def reconcile_batch_requests_in_builders(
+        self, batch_request: Union[dict, BatchRequest, RuntimeBatchRequest]
+    ) -> None:
+        """
+        Profiler "batch_request" reconciliation involves combining existing Profiler state, instantiated from Profiler configuration
+        (e.g., stored in a YAML file managed by the Profiler store), with the batch request overrides, provided at run time.
+
+        The provided batch request is propagated to the following relevant Builders attributes (as applicable):
+            - ParameterBuilders
+            - ColumnDomainBuilder
+              - We default to the latest value as a sensible default
+
+        The reconciliation logic for "batch_request" is of the "replace" nature: the provided data is consistently applied, regardless
+        of existing Builder state.
+
+        Args:
+            batch_request: Data provided at runtime used to hydrate nested builder attributes
+        """
+        if isinstance(batch_request, dict):
```

Comment:
I went the opposite way - if a dictionary, convert into a `BatchRequest` object.

I'd rather give the resulting object some context through it's type and since the builder classes take any one of dict, BatchRequest, or RuntimeBatchRequest, I think this is appropriate?

Let me know if that's not the case.

Reply:
@cdkini Happy to discuss -- I think that we need to keep builders' properties serializable.  Thanks.

Reply:
Converted to dict!